spec/unit/type/component_spec.rb

```diff
@@ -43,7 +43,7 @@
     end
 
     it "should produce the class name even for the class named main" do
-      expect(component.new(:name => "Class[main]").pathbuilder).to eq(["Main"])
+      expect(component.new(:name => "Class[main]").pathbuilder).to eq(['main'])
```

Comment:
The really funny thing about this test is that when creating the class this way (prior to this PR), the title becomes the string 'Main'. The same resource created with the symbol, does not capitalize the title. This is what happens:
```
2.3.1 :002 > x = Puppet::Resource.new('class', :main)
 => Class[main]{:name=>"main"} 
2.3.1 :003 > x.ref
 => "Class[main]" 
2.3.1 :004 > x = Puppet::Resource.new('class', 'main')
 => Class[Main]{:name=>"Main"} 
2.3.1 :005 > x.ref
 => "Class[Main]" 
```
IMHO, this doesn't make much sense.